docs/cadence/language/accounts.mdx

```diff
@@ -469,44 +400,202 @@ access(all) struct AuthAccount {
+for an account address by using the built-in `getAccount` function:
 
 ```cadence
-fun getAuthAccount(_ address: Address): AuthAccount
+fun getAccount(_ address: Address): &Account
 ```
 
-This `AuthAccount` object can perform all operations associated with authorized accounts,
-and as such this function is only available in scripts,
-which discard their changes upon completion.
-Attempting to use this function outside of a script will cause a type error.
+#### Performing write operation
+
+Access to an authorized account reference (`auth(...) &Account`)
+means having certain "write access" to it.
+
+[Entitlements](./access-control.md#entitlements) authorize access to accounts.
+Cadence provides both coarse-grained and fine-grained entitlements,
+which decide what management functions are accessible on the account.
 
-## Account Creation
+For example, the coarse-grained entitlement `Storage` grants access to all
+storage related functions, such as `save` and `load`, which save a value to storage,
+and load a value from storage respectively.
 
-Accounts can be created by calling the `AuthAccount` constructor
-and passing the account that should pay for the account creation for the `payer` parameter.
+The fine-grained entitlement `AddKey` for instance,
+grants access to only the `add` function of the `Account.Keys` value,
+that is, it grants access to adding a key to the account.
 
-The `payer` must have enough funds to be able to create an account.
+An authorized account reference like `auth(Storage, AddKey) &Account`
+therefore provides read access, as well as write access to storage,
+and the ability to add a new key to that account.
+
+[Signed transactions](./transactions.md) can get authorized account references
+for each signer of the transaction that signs as an authorizer.
+The `prepare` phase of the transaction can specify exactly which entitlements
+it needs to perform its work.
+
+For example, a transaction that deploys a contract to an account can be written as follows:
+
+```cadence
+transaction {
+    prepare(signer: auth(AddContract) &Account) {
+        signer.contracts.add(name: "MyContract", code: [/* code */])
+    }
+}
+```
+
+Here, the transaction requests an authorized reference with the `AddContract` entitlement.
+That means that the transaction is entitled to add a contract to the account,
+but is not able to add another key to the account, for example.
+
+Script can get any kind of access to any account, using the built-in `getAuthAccount` function:
+
+```cadence
+fun getAuthAccount<T: &Account>(_ address: Address): T
+```
+
+This function is only available in scripts.
+Though scripts can perform write operations,
+they discard their changes upon completion.
+Attempting to use this function outside of a script,
+for example in a transaction,
+causes a type error.
+
+## Creating an account
+
+The `Account` constructor allows creating new accounts.
+The function requires a reference to a _payer_ account,
```

Comment:
Maybe also mention the payer account reference needs (Storage | BorrowValue) entitlement

Reply:
Will do that in a follow-up to avoid conflicts